src/Compilers/Core/MSBuildTask/Utilities.cs

```diff
@@ -173,15 +167,8 @@ internal static string TryGetAssemblyPath(Assembly assembly)
                 }
                 else
                 {
-                    var callingAssembly = (Assembly)typeof(Assembly)
-                        .GetTypeInfo()
-                        .GetDeclaredMethod("GetCallingAssembly")
-                        ?.Invoke(null, null);
-
-                    var location = (string)typeof(Assembly).GetTypeInfo()
-                        .GetDeclaredProperty("Location")
-                        ?.GetMethod.Invoke(assembly, parameters: null);
-
+                    var callingAssembly = Assembly.GetCallingAssembly();
```

Comment:
>callingAssembly [](start = 24, length = 15)

Not used.

Reply:
This is a correct statement, but the semantics match the code prior to my change. Are you requesting I change the semantics by eliminating this variable?

Reply:
Should `location` be assigned `callingAssembly.Location` instead? See [https://github.com/dotnet/roslyn/pull/24239](https://github.com/dotnet/roslyn/pull/24239).

---
In reply to: [370330681](https://github.com/dotnet/roslyn/pull/41180#discussion_r370330681) [](ancestors = 370330681)

Reply:
I went ahead and removed the local variable and simplified the resulting code flow

Reply:
Waiting for @agocke before making other changes

Reply:
Probably still there because a refactoring missed it. I'm not sure this method previously took an assembly argument.